test/gosec.spawn.test.js

```diff
@@ -1,36 +1,41 @@
 // Copyright 2018 VMware, Inc.
 // SPDX-License-Identifier: BSD-2-Clause
 
-const gosec = require('../gosec/gosec')
 const fs = require('fs-extra')
 
-describe('Spawn gosec tests', () => {
-  test('Empty input for gosec', async () => {
-    const gosecResult = await gosec('', [])
-    expect(gosecResult).toBeFalsy()
-  })
+const { run } = require('../runner')
+const Gosec = require('../linters/gosec')
+
+function gosec (dir, files) {
+  return run(new Gosec(), dir, files)
+}
+
+describe('Gosec runner', () => {
+  test('Finds issues in invalid file', async () => {
+    const report = await gosec('test/fixtures/go/src/bad_files', ['bad_test_file.go'])
 
-  test('Run gosec on non go files', async () => {
-    const gosecResult = await gosec('test/fixtures/go/src/non_go_files', ['hello_world.py'])
-    expect(gosecResult).toBeFalsy()
+    expect(report.annotations.length).toEqual(6)
+    expect(report.annotations[0].start_line).toEqual('15')
+    expect(report.annotations[1].start_line).toEqual('19')
+    expect(report.annotations[2].start_line).toEqual('26')
+    expect(report.annotations[3].start_line).toEqual('27')
   })
 
-  test('Run gosec on a go file without security problems', async () => {
-    const gosecResult = await gosec('test/fixtures/go/src/secure_go_files', ['hello_world.go'])
-    expect(gosecResult.Issues.length).toEqual(0)
-    fs.remove('test/fixtures/go/src/secure_go_files/gosec.json')
+  test('Passes on correct file', async () => {
+    const report = await gosec('test/fixtures/go/src/secure_go_files', ['hello_world.go'])
+
+    expect(report.annotations).toBeFalsy()
   })
 
-  test('Run gosec on go problematic file', async () => {
-    const gosecResult = await gosec('test/fixtures/go/src/bad_files', ['bad_test_file.go'])
-    expect(gosecResult.Stats.found).toBeGreaterThan(0)
-    fs.remove('test/fixtures/go/src/bad_files/gosec.json')
+  test('Handles packages with multiple files', async () => {
+    const report = await gosec('test/fixtures/go/src/multiple_bad_files/', ['bad_test_file.go', 'networking_binding_test.go', 'randNumTest.go'])
+
+    expect(report.annotations.length).toEqual(8)
+    expect(report.annotations[4].start_line).toEqual('9')
+    expect(report.annotations[7].start_line).toEqual('16')
```

Comment:
Maybe it's a good idea to add a check how many files Gosec checked.
Because as you remembered if we place in this folder a file which ends with *_test.go then Gosec (because this is a Go rule) will think that this is a go test file and won't check it.
I mean this:
`` expect(gosecResult.Stats.files).toEqual(3)``

Reply:
I still think that check won't hurt.

Reply:
I'm not sure what you mean exactly, the fact that gosec ignores these files is outside of our control so I don't see the need to test it.

Reply:
Do we check if there are any go files when we give the folder for scan to Gosec?
I think we do with the filter function?
Because this is really important because Gosec throws an error if there are no go files in the folder.

Reply:
If there are no go files the runner returns early and generates a default report.

Reply:
Yep, you are right.